Content.Server/Construction/AnchorableSystem.cs

```diff
@@ -83,7 +83,14 @@ private void OnAnchorComplete(EntityUid uid, AnchorableComponent component, TryA
 
             // Snap rotation to cardinal (multiple of 90)
             var rot = xform.LocalRotation;
-            xform.LocalRotation = Math.Round(rot / (Math.PI / 2)) * (Math.PI / 2);
+            var snappedAngle = Math.Round(rot / (Math.PI / 2)) * Math.PI * 2;
+            // We want to override the local rotation to `snappedAngle` regardless
+            // of the xforms `NoLocalRotation` setting, so temporarily allow a
+            // local rotation and restore it
+            var oldLocalRotation = xform.NoLocalRotation;
+            xform.NoLocalRotation = false;
+            xform.LocalRotation = snappedAngle;
+            xform.NoLocalRotation = oldLocalRotation;
```

Comment:
This defeats the point of having no rotation?

Reply:
I agree that it feels kinda clumsy, but this actually _keeps_ the rotation; the issue is that if you have an object with NoLocalRotation and move it to another grid whose basis isn't aligned with the basis of the grid that the object was created in, you can never rotate that object. This at least causes the rotation to get snapped (as the code was already attempting to do) to some alignment on new grid.

Reply:
The real issue is whatever is giving it a non-zero local rotation in the first place.

Reply:
https://github.com/space-wizards/RobustToolbox/pull/3634 probably fixes this.